src/backend/distributed/commands/multi_copy.c

```diff
@@ -1217,6 +1228,209 @@ ReportCopyError(MultiConnection *connection, PGresult *result)
 }
 
 
+static void
+ConversionPathForTypes(Oid inputType, Oid destType, CopyCoercionData *result)
+{
+	Oid coercionFuncId = InvalidOid;
+	CoercionPathType coercionType;
+
+	if (destType == inputType)
+	{
+		result->coercionType = COERCION_PATH_RELABELTYPE;
+		return;
+	}
+
+	coercionType = find_coercion_pathway(destType, inputType,
+										 COERCION_EXPLICIT,
+										 &coercionFuncId);
+
+	switch (coercionType)
+	{
+		case COERCION_PATH_NONE:
+		{
+			ereport(ERROR, (errmsg("can not coerce %d to %d", inputType, destType)));
+		}
+
+		case COERCION_PATH_ARRAYCOERCE:
+		{
+			ereport(ERROR, (errmsg("can not run query which uses an implicit coercion"
+								   " between array types")));
+		}
+
+		case COERCION_PATH_COERCEVIAIO:
+		{
+			result->coercionType = COERCION_PATH_COERCEVIAIO;
+
+			{
+				bool typisvarlena = false; /* ignored */
+				Oid iofunc = InvalidOid;
+				getTypeOutputInfo(inputType, &iofunc, &typisvarlena);
+				fmgr_info(iofunc, &(result->firstCoerceFunction));
```

Comment:
here using inputFunction and outputFunction names would be nice

Reply:
The problem is that what kind of function they are changes depending on which path is being taken. For `COERCEVIAIO` they're outputFunction and inputFunction, respectively. But for `FUNC` they're just a conversion function. 

What about wasting some memory by making the struct bigger and only using the functions which are relevant to a given path?

Reply:
> What about wasting some memory by making the struct bigger and only using the functions which are relevant to a given path?

That's all I ever wanted.